s3bench.py

```diff
@@ -210,10 +216,18 @@ def list_random_objects(self):
             # generates new object's name
             object_name_given = object_analyzer.generate_object_name()
```

Comment:
In this function, we should take the prefix into consideration as this is the function that will determine the object's name. 
In this function, we'll use the `self.prefix`, and if it's not empty we'll add the given prefix to the object name and return it. 

```suggestion 
def generate_object_name(self):
        """This function generates randomized object name"""
        if self.prefix != "":
            return self.prefix + "/" + str(uuid.uuid4())
        return str(uuid.uuid4())
```